Unable to debug the project in vscode as dlls are getting blocked by some other process instantiated by vscode extension

Hi,

I am unable to debug the project because the dlls are getting blocked by some other process initiated by vscode extension. When i kill the process i can debug the code but then intellisense stop working.

PFB the error:
error MSB3021: Unable to copy file "C:\Users\user.nuget\packages\fluentassertions\4.19.4\lib\netstandard1.3\FluentAssertions.Core.dll" to "Sample\gauge_bin\FluentAssertions.Core.dll". The process cannot access the file 'Sample\gauge_bin\FluentAssertions.Core.dll' because it is being used by another process. [Sample\netcore-template.csproj]
C:\Program Files\dotnet\sdk\3.1.302\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Publish.targets(147,5): error MSB3027: Could not copy "obj\Debug\netcoreapp3.1\win-x64\netcore-template.dll" to "Sample\gauge_bin\netcore-template.dll". Exceeded r
etry count of 10. Failed. [Sample\netcore-template.csproj]
C:\Program Files\dotnet\sdk\3.1.302\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Publish.targets(147,5): error MSB3021: Unable to copy file "obj\Debug\netcoreapp3.1\win-x64\netcore-template.dll" to "Sample\gauge_bin\netcore-template.dll". The process cannot access the file 'Sample\gauge_bin\netcore-template.dll' because it is being used by another process. [Sample\netcore-template.csproj]
C:\Program Files\dotnet\sdk\3.1.302\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Publish.targets(147,5): error MSB3027: Could not copy "C:\Users\user.nuget\packages\gauge.csharp.lib\0.7.6\lib\netstandard2.0\Gauge.CSharp.Lib.dll" to "Sample\gauge_bin\Gauge.CSharp.Lib.dll". Exceeded retry count of 10. Failed. [Sample\netcore-template.csproj]
C:\Program Files\dotnet\sdk\3.1.302\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Publish.targets(147,5): error MSB3021: Unable to copy file "C:\Users\user.nuget\packages\gauge.csharp.lib\0.7.6\lib\netstandard2.0\Gauge.CSharp.Lib.dll" to "Sample\gauge_bin\Gauge.CSharp.Lib.dll". The process cannot access the file 'Sample\gauge_bin\Gauge.CSharp.Lib.dll' because it is being used by another process. [Sample\netcore-template.csproj]
{"logLevel":"fatal","message":"Unable to build Project in Sample\ndotnet Project build failed.\nRan 'dotnet publish --runtime=win-x64 --configuration=Debug --output="Sample\gauge_bin" "netcore-template.csproj" --verbosit
y=quiet'\n at Gauge.Dotnet.GaugeProjectBuilder.BuildTargetGaugeProject()\r\n at Gauge.Dotnet.StartCommand.TryBuild()"}

Gauge version: 1.1.3
Commit Hash: 9d3a3d5

Plugins
-------
dotnet (0.1.7)
html-report (4.0.8)
screenshot (0.0.1)

--- a/src/GaugeListener.cs
+++ b/src/GaugeListener.cs
@@ -31,17 +31,10 @@ namespace Gauge.Dotnet
                 var reflectionWrapper = new ReflectionWrapper();
                 var activatorWrapper = new ActivatorWrapper();
                 Logger.Debug($"Loading assembly from : {assemblyPath}");
-                if (string.Compare(Environment.GetEnvironmentVariable("IS_DAEMON"), "true", true) == 0)
+                var isDaemon = string.Compare(Environment.GetEnvironmentVariable("IS_DAEMON"), "true", true) == 0;
+                using (var gaugeLoadContext = new GaugeLoadContext(assemblyPath, isDaemon))
                 {
-                    using (var gaugeLoadContext = new DisposableGaugeLoadContext(assemblyPath))
-                    {
-                        var assemblyLoader = new AssemblyLoader(assemblyPath, gaugeLoadContext, reflectionWrapper, activatorWrapper, _staticLoader.GetStepRegistry());
-                        var handler = new RunnerServiceHandler(activatorWrapper,reflectionWrapper, assemblyLoader, _staticLoader, server);
-                        server.Services.Add(Runner.BindService(handler));
-                    }
-                } else {
-                    var assemblyLoader = new AssemblyLoader(assemblyPath, new GaugeLoadContext(assemblyPath), reflectionWrapper,
-                        activatorWrapper, _staticLoader.GetStepRegistry());
+                    var assemblyLoader = new AssemblyLoader(assemblyPath, gaugeLoadContext, reflectionWrapper, activatorWrapper, _staticLoader.GetStepRegistry());
                     var handler = new RunnerServiceHandler(activatorWrapper,reflectionWrapper, assemblyLoader, _staticLoader, server);
                     server.Services.Add(Runner.BindService(handler));
                 }

--- a/src/GaugeListener.cs
+++ b/src/GaugeListener.cs
@@ -31,10 +31,11 @@ namespace Gauge.Dotnet
                 var reflectionWrapper = new ReflectionWrapper();
                 var activatorWrapper = new ActivatorWrapper();
                 Logger.Debug($"Loading assembly from : {assemblyPath}");
-                var gaugeLoadContext = new GaugeLoadContext(assemblyPath);
-                var assemblyLoader = new AssemblyLoader(assemblyPath, gaugeLoadContext, reflectionWrapper, activatorWrapper, _staticLoader.GetStepRegistry());
-                var handler = new RunnerServiceHandler(activatorWrapper,reflectionWrapper, assemblyLoader, _staticLoader, server);
-                server.Services.Add(Runner.BindService(handler));
+                using (var gaugeLoadContext = new GaugeLoadContext(assemblyPath)) {
+                    var assemblyLoader = new AssemblyLoader(assemblyPath, gaugeLoadContext, reflectionWrapper, activatorWrapper, _staticLoader.GetStepRegistry());
+                    var handler = new RunnerServiceHandler(activatorWrapper,reflectionWrapper, assemblyLoader, _staticLoader, server);
+                    server.Services.Add(Runner.BindService(handler));
+                }
             } else {
                 var handler = new RunnerServiceHandler(_staticLoader, server);
                 server.Services.Add(Runner.BindService(handler));

--- a/src/GaugeListener.cs
+++ b/src/GaugeListener.cs
@@ -31,8 +31,17 @@ namespace Gauge.Dotnet
                 var reflectionWrapper = new ReflectionWrapper();
                 var activatorWrapper = new ActivatorWrapper();
                 Logger.Debug($"Loading assembly from : {assemblyPath}");
-                using (var gaugeLoadContext = new GaugeLoadContext(assemblyPath)) {
-                    var assemblyLoader = new AssemblyLoader(assemblyPath, gaugeLoadContext, reflectionWrapper, activatorWrapper, _staticLoader.GetStepRegistry());
+                if (string.Compare(Environment.GetEnvironmentVariable("IS_DAEMON"), "true", true) == 0)
+                {
+                    using (var gaugeLoadContext = new DisposableGaugeLoadContext(assemblyPath))
+                    {
+                        var assemblyLoader = new AssemblyLoader(assemblyPath, gaugeLoadContext, reflectionWrapper, activatorWrapper, _staticLoader.GetStepRegistry());
+                        var handler = new RunnerServiceHandler(activatorWrapper,reflectionWrapper, assemblyLoader, _staticLoader, server);
+                        server.Services.Add(Runner.BindService(handler));
+                    }
+                } else {
+                    var assemblyLoader = new AssemblyLoader(assemblyPath, new GaugeLoadContext(assemblyPath), reflectionWrapper,
+                        activatorWrapper, _staticLoader.GetStepRegistry());
                     var handler = new RunnerServiceHandler(activatorWrapper,reflectionWrapper, assemblyLoader, _staticLoader, server);
                     server.Services.Add(Runner.BindService(handler));
                 }

--- a/src/GaugeLoadContext.cs
+++ b/src/GaugeLoadContext.cs
@@ -11,7 +11,7 @@ using System.Runtime.Loader;
 
 namespace Gauge.Dotnet
 {
-    public class GaugeLoadContext: AssemblyLoadContext, IGaugeLoadContext, System.IDisposable
+    public class GaugeLoadContext: AssemblyLoadContext, IGaugeLoadContext
     {
         private const string GaugeLibAssemblyName = "Gauge.CSharp.Lib";
         private AssemblyDependencyResolver _resolver;
@@ -21,10 +21,9 @@ namespace Gauge.Dotnet
             _resolver = new AssemblyDependencyResolver(pluginPath);
         }
 
-        public void Dispose()
+        protected GaugeLoadContext(string pluginPath, bool collectible) : base(collectible)
         {
-            Logger.Debug("Unloading GaugeLoadContext, assemblies should be freed up.");
-            base.Unload();
+            _resolver = new AssemblyDependencyResolver(pluginPath);
         }
 
         public IEnumerable<Assembly> GetAssembliesReferencingGaugeLib()
@@ -42,5 +41,19 @@ namespace Gauge.Dotnet
             }
             return null;
         }
+    }
+
+    public class DisposableGaugeLoadContext : GaugeLoadContext, System.IDisposable
+    {
+        public DisposableGaugeLoadContext(string pluginPath) : base(pluginPath, true)
+        {
+        }
+
+        public void Dispose()
+        {
+            Logger.Debug("Unloading GaugeLoadContext, assemblies should be freed up.");
+            base.Unload();
+        }
+
     }
 }

--- a/src/GaugeLoadContext.cs
+++ b/src/GaugeLoadContext.cs
@@ -11,13 +11,20 @@ using System.Runtime.Loader;
 
 namespace Gauge.Dotnet
 {
-    public class GaugeLoadContext: AssemblyLoadContext, IGaugeLoadContext {
+    public class GaugeLoadContext: AssemblyLoadContext, IGaugeLoadContext, System.IDisposable
+    {
         private const string GaugeLibAssemblyName = "Gauge.CSharp.Lib";
         private AssemblyDependencyResolver _resolver;
 
         public GaugeLoadContext(string pluginPath)
         {
             _resolver = new AssemblyDependencyResolver(pluginPath);
+        }
+
+        public void Dispose()
+        {
+            Logger.Debug("Unloading GaugeLoadContext, assemblies should be freed up.");
+            base.Unload();
         }
 
         public IEnumerable<Assembly> GetAssembliesReferencingGaugeLib()

--- a/src/GaugeLoadContext.cs
+++ b/src/GaugeLoadContext.cs
@@ -11,19 +11,23 @@ using System.Runtime.Loader;
 
 namespace Gauge.Dotnet
 {
-    public class GaugeLoadContext: AssemblyLoadContext, IGaugeLoadContext
+    public class GaugeLoadContext: AssemblyLoadContext, IGaugeLoadContext, System.IDisposable
     {
         private const string GaugeLibAssemblyName = "Gauge.CSharp.Lib";
         private AssemblyDependencyResolver _resolver;
 
-        public GaugeLoadContext(string pluginPath)
+        public GaugeLoadContext(string pluginPath, bool collectibe = false) : base(collectibe)
         {
             _resolver = new AssemblyDependencyResolver(pluginPath);
         }
 
-        protected GaugeLoadContext(string pluginPath, bool collectible) : base(collectible)
+        public void Dispose()
         {
-            _resolver = new AssemblyDependencyResolver(pluginPath);
+            if (IsCollectible)
+            {
+                Logger.Debug("Unloading GaugeLoadContext, assemblies should be freed up.");
+                base.Unload();
+            }
         }
 
         public IEnumerable<Assembly> GetAssembliesReferencingGaugeLib()
@@ -41,19 +45,5 @@ namespace Gauge.Dotnet
             }
             return null;
         }
-    }
-
-    public class DisposableGaugeLoadContext : GaugeLoadContext, System.IDisposable
-    {
-        public DisposableGaugeLoadContext(string pluginPath) : base(pluginPath, true)
-        {
-        }
-
-        public void Dispose()
-        {
-            Logger.Debug("Unloading GaugeLoadContext, assemblies should be freed up.");
-            base.Unload();
-        }
-
     }
 }

--- a/src/GaugeLoadContext.cs
+++ b/src/GaugeLoadContext.cs
@@ -16,7 +16,7 @@ namespace Gauge.Dotnet
         private const string GaugeLibAssemblyName = "Gauge.CSharp.Lib";
         private AssemblyDependencyResolver _resolver;
 
-        public GaugeLoadContext(string pluginPath, bool collectibe = false) : base(collectibe)
+        public GaugeLoadContext(string pluginPath, bool collectible = false) : base(collectible)
         {
             _resolver = new AssemblyDependencyResolver(pluginPath);
         }

